Dockerfile.etl

```diff
@@ -0,0 +1,45 @@
+# syntax=docker/dockerfile:1
+FROM python:3.10.7
+WORKDIR /code
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# if you use an alpine python image, the following is necessary
+# RUN apk add --no-cache gcc musl-dev linux-headers
+# Note: This version of poetry must match what is in the `flake.nix`
+RUN pip install "poetry==1.3.2"
+
+RUN python -m venv /venv
+
+# Install dependencies first for better caching
+RUN mkdir -p ./build/api
+COPY api/pyproject.toml build/api/pyproject.toml
+COPY api/poetry.lock build/api/poetry.lock
+RUN poetry --directory build/api export -f requirements.txt | /venv/bin/pip install -r /dev/stdin
+
+# install nix for data extraction
+RUN curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install linux --init none --no-confirm
+ENV PATH="${PATH}:/nix/var/nix/profiles/default/bin"
```

Comment:
This should run earlier in the build as it is less likely to change than Python deps used by the project.

Usually it goes like this in a Dockerfile
- set environment variables
- instal system dependencies
- install project dependencies
- install project package itself